### PR TITLE
Fix MacOS compile error

### PR DESCRIPTION
**Pull request by max5800**
Related [Issue 206](https://github.com/BeamMP/BeamMP-Server/issues/206)

### CODE REVIEW

**Comment by lionkor** [comment]
I approved the CI/CD run on this, if it runs through I'll review + merge :) Thank you very much!

File: src/TLuaEngine.cpp [hunk 1]
```diff
@@ -268,7 +268,7 @@ std::vector<std::string> TLuaEngine::StateThreadData::GetStateTableKeys(const st
 
     for (size_t i = 0; i < keys.size(); ++i) {
         auto obj = current.get<sol::object>(keys.at(i));
-        if (obj.get_type() == sol::type::nil) {
+        if (obj.get_type() == sol::type::none) {
```
Comment (lionkor) [inline src/TLuaEngine.cpp]:
```suggestion
        if (obj.get_type() == sol::type::lua_nil) {
```

none is a different type (LUA_TNONE) signifying no type, whereas lua_nil is the proper way to do `nil`

**Review by lionkor — CHANGES_REQUESTED** [review]

File: src/TLuaEngine.cpp (hunk 1 above)
Comment (max5800) [reply src/TLuaEngine.cpp]: Alright, I will test that later when I'm back at home.

Comment (max5800) [reply src/TLuaEngine.cpp]: changed to "lua_nil"  -> got successfull compiles on macos and my debian container. :)

**Review by lionkor — APPROVED** [review]
lgtm thank you very much

**Comment by max5800** [comment]
You are welcome, it was fun!
Would be my first merged pull request. 😄🎉

**Comment by lionkor** [comment]
Thanks, will be released in the next update

**Pull request merged**